packages/frontend-2/components/header/NavUserMenu.vue

```diff
@@ -160,32 +185,38 @@ const { logout } = useAuthManager()
 const { activeUser, isGuest } = useActiveUser()
 const { isDarkTheme, toggleTheme } = useTheme()
 const { serverInfo } = useServerInfo()
-const router = useRouter()
+
 const route = useRoute()
 
 const showInviteDialog = ref(false)
-const showProfileEditDialog = ref(false)
+const showSettingsDialog = ref(false)
+const settingsRoute = ref('')
+const settingsOriginalRoute = ref('')
+const breakpoints = useBreakpoints(TailwindBreakpoints)
+const isMobile = breakpoints.smallerOrEqual('md')
 const menuButtonId = useId()
 
 const Icon = computed(() => (isDarkTheme.value ? SunIcon : MoonIcon))
 const version = computed(() => serverInfo.value?.version)
 const isAdmin = computed(() => activeUser.value?.role === Roles.Server.Admin)
-const isProfileRoute = computed(() => route.path === profileRoute)
+const isSettingsRoute = computed(() => !!getSettingsRoute(route.path))
 
 const toggleInviteDialog = () => {
   showInviteDialog.value = true
 }
 
-const goToServerManagement = () => {
-  router.push('/server-management')
+const toggleSettingsDialog = (newRoute: string) => {
+  showSettingsDialog.value = true
+  settingsRoute.value = newRoute
+  settingsOriginalRoute.value = route.path
 }
 
 watch(
-  isProfileRoute,
+  isSettingsRoute,
```

Comment:
refactor all of this route stuff, as described yesterday. once that's done, feel free to re-request the review and I'll go through everything